### ISSUE TITLE
[upgrade] Projecting nested structural directives breaks

### ISSUE THREAD

**Issue by youdz** [angular/angular#14260]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

When projecting AngularJS content inside of a downgraded Angular component, any nested structural directive in the AngularJS content will not render. The top-level structural directives work, but the ones nested under them don't.

**Expected behavior**

Structural directives should render properly no matter their place in the projected content.

**Minimal reproduction of the problem with instructions**

Simple nested ng-repeat on a dummy component:
http://plnkr.co/edit/IY48pPtocWia8ZbWHvVY?p=preview

Also note that this **used to work**, see this 2.3.0 version of the exact same app:
http://plnkr.co/edit/zRq1uiPDBkmUQXP5JJ4A?p=preview

**What is the motivation / use case for changing the behavior?**

Both reverting a breaking change that was not documented, and solve the breakage of structural directives.

* **Angular version:** 2.4.X

* **Browser:** all

 
* **Language:** all

**Comment by petebacondarwin** [comment]
This is fixed in master: http://plnkr.co/edit/LaJfI6AjdvpnT5duWQgD?p=preview
I need to look to see if it can be easily backported to 2.4.x

**Comment by petebacondarwin** [comment]
@gkalpak reports that this is a simple fix:

> (We are compiling the wrong element's contents (`templateElement.contents()` instead of `element.contents()`): https://github.com/angular/angular/blob/2.4.x/modules/%40angular/upgrade/src/upgrade_adapter.ts#L674)

**Comment by gkalpak** [comment]
Fixed on 2.4.x (with 74cb57521).
